apollo-router/src/orbiter/mod.rs

```diff
@@ -0,0 +1,392 @@
+use serde::Serialize;
+use serde_json::Map;
+use serde_json::Value;
+use tower::BoxError;
+use uuid::Uuid;
+
+use crate::configuration::generate_config_schema;
+// With regards to ELv2 licensing, this entire file is license key functionality
+use crate::executable::Opt;
+use crate::plugin::DynPlugin;
+use crate::router_factory::RouterSuperServiceFactory;
+use crate::spec::Schema;
+use crate::Configuration;
+
+/// This session id is created once when the router starts. It persists between config reloads.
+static SESSION_ID: OnceCell<Uuid> = OnceCell::new();
+
+/// Platform represents the platform the CLI is being run from
+#[derive(Debug, Serialize)]
+struct Platform {
+    /// the platform from which the command was run (i.e. linux, macOS, windows or even wsl)
+    os: String,
+
+    /// if we think this command is being run in CI
+    continuous_integration: Option<ci_info::types::Vendor>,
+}
+
+/// A usage report for the router
+#[derive(Serialize)]
+struct UsageReport {
+    /// A random ID that is generated on first startup of the Router. It is not persistent between restarts of the Router, but will be persistent for hot reloads
+    session_id: Uuid,
+    /// The version of the Router
+    version: String,
+    /// Information about the current architecture/platform
+    platform: Platform,
+    /// Information about what was being used
+    usage: Map<String, Value>,
+}
+
+impl<T: RouterSuperServiceFactory> OrbiterRouterSuperServiceFactory<T> {
+    pub(crate) fn new(delegate: T) -> OrbiterRouterSuperServiceFactory<T> {
+        OrbiterRouterSuperServiceFactory { delegate }
+    }
+}
+
+/// A service factory that will report some anonymous telemetry to Apollo. It can be disabled by users, but the data is useful for helping us to decide where to spend our efforts.
+/// In future we should try and move this towards otel metrics, this will allow us to send the information direct to something that ingests OTLP.
+/// The data sent looks something like this:
+/// ```json
+/// {
+///   "session_id": "fbe09da3-ebdb-4863-8086-feb97464b8d7",
+///   "version": "1.4.0", // The version of the router
+///   "os": "linux",
+///   "ci": null,
+///   "usage": {
+///     "configuration.headers.all.request.propagate.named.redacted": 3
+///     "configuration.headers.all.request.propagate.default.redacted": 1
+///     "configuration.headers.all.request.len": 3
+///     "configuration.headers.subgraphs.redacted.request.propagate.named.redacted": 2
+///     "configuration.headers.subgraphs.redacted.request.len": 2
+///     "configuration.headers.subgraphs.len": 1
+///     "configuration.homepage.enabled.true": 1
+///     "args.config-path.redacted": 1,
+///     "args.hot-reload.true": 1,
+///     //Many more keys. This is dynamic and will change over time.
+///     //More...
+///     //More...
+///     //More...
+///   }
+/// }
+/// ```
+#[derive(Default)]
+pub(crate) struct OrbiterRouterSuperServiceFactory<T: RouterSuperServiceFactory> {
```

Comment:
Does this need to be a struct or is it really only an extension trait? It only has a delegate, and that (by definition) will be available as self in the base struct.
(NB: Hard to make this kind of comment with total confidence in review, but it just feels like that)

Reply:
I don't think that this can be an extension trait as the struct implementation on YamlRouterFactory would always win.
Let's leave it for now.